src/test/architecture/test_architecture.py

```diff
@@ -1019,3 +1019,140 @@ def test_rename_path(self, mocker):
         assert set(arch.communications) == set(arch_expected.communications)
         assert set(arch.paths) == set(arch_expected.paths)
         assert set(arch.executors) == set(arch_expected.executors)
+
+
+class TestDiff:
+
+    def test_diff_node_names(self, mocker):
+        architecture_mock1 = mocker.Mock(spec=Architecture)
+        architecture_mock2 = mocker.Mock(spec=Architecture)
+        mocker.patch.object(architecture_mock1,
+                            'node_names',
+                            ('drive_node', 'actuator_dummy_node'))
+        mocker.patch.object(architecture_mock2,
+                            'node_names',
+                            ('drive_node', 'filter_node', 'message_driven_node'))
+        name1, name2 = Architecture.diff_node_names(architecture_mock1, architecture_mock2)
```

Comment:
For the entire test code:

The return value of the function is multiple names of node .
Please make sure that this is evident from the naming.

An example is as follows:
```suggestion
        node_names_in_mock1, node_names_in_mock2 = Architecture.diff_node_names(architecture_mock1, architecture_mock2)
```